lib/article_iterator.rb

```diff
@@ -0,0 +1,44 @@
+require 'date'
+
+class ArticleIterator
```

Comment:
I needed a helper that could walk the `_posts` directory and give me all the posts in batches.